path parsed starts with a double quote ("), and the parser expects a raw path/URI instead during Junit test on Windows machine

### Describe the bug

 Cause: org.junit.platform.commons.PreconditionViolationException: Could not load class with name: accept.CreateExportFilesIT
[ERROR] 	at org.junit.platform.engine.discovery.ClassSelector.lambda$getJavaClass$0(ClassSelector.java:100)
[ERROR] 	at org.junit.platform.commons.function.Try$Failure.getOrThrow(Try.java:379)
[ERROR] 	at org.junit.platform.commons.function.Try.getNonNullOrThrow(Try.java:199)
[ERROR] 	at org.junit.platform.engine.discovery.ClassSelector.getJavaClass(ClassSelector.java:99)
[ERROR] 	at org.junit.jupiter.engine.discovery.ClassSelectorResolver.resolve(ClassSelectorResolver.java:86)
[ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.lambda$resolve$0(EngineDiscoveryRequestResolution.java:137)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:214)
[ERROR] 	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1693)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:147)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:588)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:574)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
[ERROR] 	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:683)
[ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:191)
[ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:128)
[ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolveCompletely(EngineDiscoveryRequestResolution.java:94)
[ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.run(EngineDiscoveryRequestResolution.java:85)
[ERROR] 	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolver.resolve(EngineDiscoveryRequestResolver.java:150)
[ERROR] 	at org.junit.jupiter.engine.discovery.DiscoverySelectorResolver.resolveSelectors(DiscoverySelectorResolver.java:62)
[ERROR] 	at org.junit.jupiter.engine.JupiterTestEngine.discover(JupiterTestEngine.java:74)
[ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:195)
[ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverSafely(EngineDiscoveryOrchestrator.java:174)
[ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:119)
[ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:84)
[ERROR] 	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:119)
[ERROR] 	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:85)
[ERROR] 	at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:43)
[ERROR] 	at org.junit.platform.launcher.core.InterceptingLauncher.lambda$discover$0(InterceptingLauncher.java:35)
[ERROR] 	at org.junit.platform.launcher.core.ClasspathAlignmentCheckingLauncherInterceptor.intercept(ClasspathAlignmentCheckingLauncherInterceptor.java:25)
[ERROR] 	at org.junit.platform.launcher.core.InterceptingLauncher.discover(InterceptingLauncher.java:35)
[ERROR] 	at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:43)
[ERROR] 	at org.apache.maven.surefire.junitplatform.LauncherAdapter.discover(LauncherAdapter.java:47)
[ERROR] 	at org.apache.maven.surefire.junitplatform.TestPlanScannerFilter.accept(TestPlanScannerFilter.java:51)
[ERROR] 	at org.apache.maven.surefire.api.util.DefaultScanResult.applyFilter(DefaultScanResult.java:87)
[ERROR] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.scanClasspath(JUnitPlatformProvider.java:156)
[ERROR] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:136)
[ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
[ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
[ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
[ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
[ERROR] Caused by: java.nio.file.InvalidPathException: Illegal char <"> at index 0: "C:\\..\\..\\..\\...\\target\\quarkus\\bootstrap\\test-app-model.dat"
[ERROR] 	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:191)
[ERROR] 	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:142)
[ERROR] 	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:46)
[ERROR] 	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
[ERROR] 	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:203)
[ERROR] 	at java.base/java.nio.file.Path.of(Path.java:148)
[ERROR] 	at java.base/java.nio.file.Paths.get(Paths.java:69)
[ERROR] 	at io.quarkus.bootstrap.BootstrapAppModelFactory.loadFromSystemProperty(BootstrapAppModelFactory.java:290)
[ERROR] 	at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModel(BootstrapAppModelFactory.java:202)
[ERROR] 	at io.quarkus.test.junit.AppMakerHelper.makeCuratedApplication(AppMakerHelper.java:151)
[ERROR] 	at io.quarkus.test.junit.classloading.FacadeClassLoader.getOrCreateCuratedApplication(FacadeClassLoader.java:551)
[ERROR] 	at io.quarkus.test.junit.classloading.FacadeClassLoader.getOrCreateRuntimeClassLoader(FacadeClassLoader.java:581)
[ERROR] 	at io.quarkus.test.junit.classloading.FacadeClassLoader.getQuarkusClassLoader(FacadeClassLoader.java:481)
[ERROR] 	at io.quarkus.test.junit.classloading.FacadeClassLoader.loadClass(FacadeClassLoader.java:365)
[ERROR] 	at java.base/java.lang.Class.forName0(Native Method)
[ERROR] 	at java.base/java.lang.Class.forName

-------------------------------------------------------------------------------------------------------------------
Caused by: java.nio.file.InvalidPathException: Illegal char <"> at index 0: "C:\\...\\...\\..\\....r\\target\\quarkus\\bootstrap\\test-app-model.dat"
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:191)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:142)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:46)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:203)
	at java.base/java.nio.file.Path.of(Path.java:148)
	at java.base/java.nio.file.Paths.get(Paths.java:69)
	at io.quarkus.bootstrap.BootstrapAppModelFactory.loadFromSystemProperty(BootstrapAppModelFactory.java:290)
---------------------------------------------

Maven output:

[INFO] --- jacoco:0.8.14:prepare-agent-integration (agent-for-it) @ xxxxxxxx ---
argLine set to -javaagent:....  --add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-exports=java.base/jdk.internal.module=ALL-UNNAMED "-Dquarkus-internal-test.serialized-app-model.path=**\"C:\\ontwikkel\\git\\claims\\cis-claim-exporter\\target\\quarkus\\bootstrap\\test-app-model.dat\""**

### Expected behavior

Should not have an URI between double quotes. 

### Actual behavior

Has quotes in the path

### How to Reproduce?

needed?

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.26200.8246]

### Output of `java -version`

openjdk 25.0.1 2025-10-21

### Quarkus version or git rev

3.34.5

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.13 (39d686bd50d8e054301e3a68ad44781df6f80dda)

### Additional information

_No response_

--- a/devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java
@@ -126,11 +126,14 @@ public class GenerateCodeMojo extends QuarkusBootstrapMojo {
 
                             //Pass the application model to the Surefire/Failsafe test process as a system property,
                             //so it doesn't have to do a workspace search to find it. Same as we do for Gradle.
+                            // Use a relative path to avoid quoting issues with spaces and backslashes
+                            // in absolute paths on Windows when passed through Surefire's argLine tokenizer.
+                            Path relativeAppModelPath = baseDir().toPath().relativize(serializedTestAppModelPath);
                             Properties properties = mavenProject().getProperties();
                             String argLine = properties.getProperty("argLine", "");
                             properties.setProperty("argLine", argLine +
-                                    " -D" + BootstrapConstants.SERIALIZED_TEST_APP_MODEL + "=\"" + serializedTestAppModelPath
-                                    + "\"");
+                                    " -D" + BootstrapConstants.SERIALIZED_TEST_APP_MODEL + "="
+                                    + relativeAppModelPath);
                         } catch (IOException e) {
                             getLog().warn("Failed to serialize application model", e);
                         }

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java
@@ -287,7 +287,10 @@ public class BootstrapAppModelFactory {
         final String serializedModel = test ? System.getProperty(BootstrapConstants.SERIALIZED_TEST_APP_MODEL)
                 : System.getProperty(BootstrapConstants.SERIALIZED_APP_MODEL);
         if (serializedModel != null) {
-            final Path p = Paths.get(serializedModel);
+            Path p = Paths.get(serializedModel);
+            if (!p.isAbsolute() && projectRoot != null) {
+                p = projectRoot.resolve(p);
+            }
             if (Files.exists(p)) {
                 try {
                     return new CurationResult(ApplicationModelSerializer.deserialize(p));

